sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,497 @@
+
+/**
+ * Check if low power state is enabled.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_state Whether the power manager will transition to low power
+ *                  state on the next WFI instruction.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_enabled_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_toggle_t *cur_state);
+
+/**
+ * Low power state configuration.
+ *
+ * Specifies whether various components should be enabled in low power state.
+ */
+typedef struct dif_pwrmgr_low_power_config {
+  /**
+   * Whether the core clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t core_clock;
+  /**
+   * Whether the input/output (IO) clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t io_clock;
+  /**
+   * Whether the main power domain should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t main_power_domain;
+} dif_pwrmgr_low_power_config_t;
+
+/**
+ * Configure low power state.
+ *
+ * @param pwrmgr A power manager.
+ * @param new_config New low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_set(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *new_config);
+
+/**
+ * Get the current low power state configuration.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_config Current low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *cur_config);
+
+/**
+ * Request types.
+ */
+typedef enum dif_pwrmgr_req_type {
+  /**
+   * A wake-up request.
+   */
+  kDifPwrmgrReqTypeWakeUp,
+  /**
+   * A reset request.
+   */
+  kDifPwrmgrReqTypeReset,
+} dif_pwrmgr_req_type_t;
+
+/**
+ * Lock sources for a request type.
+ *
+ * Once sources for a request type is locked, they cannot be changed
+ * until the chip is reset.
+ *
+ * @param pwrmgr A power manager.
+ * @param req_type A request type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_request_sources_lock(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_req_type_t req_type);
+
+/**
+ * Check if sources for a request type is locked.
+ *
+ * @param pwrmgr A power manager.
+ * @param req_type A request type.
+ * @param is_locked Whether sources for the request type is locked.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_request_sources_is_locked(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_req_type_t req_type,
+    dif_pwrmgr_toggle_t *is_locked);
+
+/**
+ * Return codes for functions that write to configuration registers.
+ */
+typedef enum dif_pwrmgr_config_result {
+  /**
+   * The call succeeded.
+   */
+  kDifPwrmgrConfigResultOK = 0,
```

Comment:
Thoughts on using `kDifPwrmgrConfigResultOk = kDifPwrmgrOk` style throughout?

Reply:
I think this is what is supposed to be used.

Reply:
I agree, thanks for catching this!